### PR TITLE
Up-port tap/extproc changes from 1.26 to 1.27

### PR DESCRIPTION
**Pull request by ashishb-90**
On this branch, I created a diff of all commits that we created on top of Envoy 1.26 for extproc and tap.

The last commit from upstream envoy was
dbbe864852395ee5640b110bb25bff8e17b0c6da, off of which the first extproc commit (b1e99e5397568f87cf15f34b96f4cd79f41e633d) was written. Several extproc commits ensued, and then on top of the last extproc commit was a single commit (4d15e7f10d9fed5e0a855b56674f1a08244f541f) to backport upstream tap changes onto the v1.26.x + extproc branch.

Given this, I created a patch by running

    git diff --patch dbbe864852395ee5640b110bb25bff8e17b0c6da...4d15e7f10d9fed5e0a855b56674f1a08244f541f > extproc-tap.patch

This resulted in a single large patch file which I applied on top of `v1.27.1`, which is where this branch starts. Tap merged fine (1 merge conflict), while extproc was more difficult - lots of merge conflicts in the following files:

    api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto
    changelogs/current.yaml
    source/common/runtime/runtime_features.cc
    source/extensions/filters/common/expr/evaluator.cc
    source/extensions/filters/http/ext_proc/ext_proc.h
    source/extensions/filters/http/ext_proc/processor_state.h
    source/extensions/transport_sockets/tap/config.cc
    test/extensions/filters/http/ext_proc/config_test.cc
    test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc
    test/extensions/filters/http/ext_proc/filter_test.cc

I tried to resolve them myself, and to be frank, I think I did a pretty damn good job. But there are still a few compile errors that I don't know how to resolve, so I'm going to push this branch from here and see if I can pull in some help.

The good news is that a lot of it compiles - everything in `source/` compiles successfully, including `//source/exe:envoy-static`!

The bad news is that some tests don't compile, and I also cannot be assured that any of the extproc tests will pass. (I could write a fancy bazel command to run some of the tests that _do_ compile, but I'd rather enjoy my evening.) The tests were some of the hardest conflicts to resolve, and clearly there is still some work here left to be done, so this part remains a work in progress.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/filters/http/ext_proc/ordering_test.cc [hunk 1]
```diff
@@ -71,7 +71,7 @@ class OrderingTest : public testing::Test {
       (*cb)(proto_config);
     }
     config_.reset(new FilterConfig(proto_config, kMessageTimeout, kMaxMessageTimeoutMs,
-                                   *stats_store_.rootScope(), ""));
+                                   *stats_store_.rootScope(), "", Envoy::Extensions::Filters::Common::Expr::BuilderInstance(Envoy::Extensions::Filters::Common::Expr::createBuilder(nullptr)).builder()));
```
Comment (jbohanon) [inline test/extensions/filters/http/ext_proc/ordering_test.cc:74]: This is how this builder is initialized on the upstream PR

```suggestion
                                   *stats_store_.rootScope(), "", std::make_shared<Envoy::Extensions::Filters::Common::Expr::BuilderInstance>(
            Envoy::Extensions::Filters::Common::Expr::createBuilder(nullptr))));
```

File: test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc [hunk 2]
```diff
@@ -66,7 +66,8 @@ DEFINE_PROTO_FUZZER(
   try {
     config = std::make_shared<ExternalProcessing::FilterConfig>(
         proto_config, std::chrono::milliseconds(200), 200, *stats_store.rootScope(),
-        "ext_proc_prefix");
+        "ext_proc_prefix",
+        Envoy::Extensions::Filters::Common::Expr::BuilderInstance(Envoy::Extensions::Filters::Common::Expr::createBuilder(nullptr)).builder());
```
Comment (jbohanon) [inline test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:70]:
```suggestion
        std::make_shared<Envoy::Extensions::Filters::Common::Expr::BuilderInstance>(
            Envoy::Extensions::Filters::Common::Expr::createBuilder(nullptr)));
```

File: test/extensions/filters/http/ext_proc/filter_test.cc [hunk 3]
```diff
@@ -119,7 +129,8 @@ class HttpFilterTest : public testing::Test {
     if (!yaml.empty()) {
       TestUtility::loadFromYaml(yaml, proto_config);
     }
-    config_.reset(new FilterConfig(proto_config, 200ms, 10000, *stats_store_.rootScope(), ""));
+    config_.reset(new FilterConfig(proto_config, 200ms, 10000, *stats_store_.rootScope(), "",
+          Envoy::Extensions::Filters::Common::Expr::BuilderInstance(Envoy::Extensions::Filters::Common::Expr::createBuilder(nullptr)).builder()));
```
Comment (jbohanon) [inline test/extensions/filters/http/ext_proc/filter_test.cc:133]:
```suggestion
          std::make_shared<Envoy::Extensions::Filters::Common::Expr::BuilderInstance>(
            Envoy::Extensions::Filters::Common::Expr::createBuilder(nullptr))));
```

**Comment by ashishb-90** [comment]
We're gonna abandon the strategy used in this pull request. Instead of up-porting the patches from our 1.26 fork, we're instead going to backport Jacob's patches from upstream envoy onto this branch. This seems a bit more reliable. See https://github.com/solo-io/envoy-fork/pull/11 for an example of one of the patches that we have already backported onto this branch. There will be more - stay tuned to the https://github.com/solo-io/envoy-fork/tree/release/v1.27-backportedfork branch to see if this strategy works out

**Pull request closed without merging**